src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -874,10 +874,9 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                 AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeNullableContextAttribute(this, nullableContextValue));
             }
 
-            bool isCovariantOverride = this.IsOverride && !this.OverriddenMethod.ReturnType.Equals(this.ReturnType, TypeCompareKind.AllIgnoreOptions);
-            if (isCovariantOverride)
+            if (this.RequiresExplicitOverride(out _))
```

Comment:
>this.RequiresExplicitOverride(out _) [](start = 16, length = 36)

I do not believe this is we would want to apply the attribute across the board (especially for cases for which we emitted MethodImpl in previous versions of the compiler), but rather should target true covariant return cases. I also think that for those, the attribute application is not optional, but required.

Reply:
This PR implements [https://github.com/dotnet/roslyn/issues/45372](https://github.com/dotnet/roslyn/issues/45372). This is intentional. The behavior of the methodimpl in the absence of this attribute does not match the language semantics, particularly in binary compat scenarios. On advice of David Wrighton, when it is available this attribute is used on all methodimpl for a class member override as described in [https://github.com/dotnet/roslyn/issues/45372](https://github.com/dotnet/roslyn/issues/45372).

---
In reply to: [452991537](https://github.com/dotnet/roslyn/pull/45870#discussion_r452991537) [](ancestors = 452991537)

Reply:
What to do when the attribute is missing is an open issue recorded in [https://github.com/dotnet/roslyn/issues/44208](https://github.com/dotnet/roslyn/issues/44208)

---
In reply to: [453862057](https://github.com/dotnet/roslyn/pull/45870#discussion_r453862057) [](ancestors = 453862057,452991537)